squirrel/sqpcheader.h

```diff
@@ -6,6 +6,17 @@
 #include <crtdbg.h>
 #endif
 
+#if defined(_MSC_VER) 
+#pragma warning(disable: 4706) // assignment within conditional expression 
+#pragma warning(disable: 4611) // interaction between '_setjmp' and C++ object destruction is non-portable
+#endif
+
+#if defined(_MSC_VER) 
```

Comment:
Enable platform/compiler specific handling for unreachable code.

The _MSC_VER condition could be changed to a user defined flag, allowing
users to choose the behaviour of UNREACHABLE.

Happy to accept feedback.